### PR TITLE
Examples for STM32_Smart v2.0 development board.

### PR DESCRIPTION
**Pull request by baranovskiykonstantin**
All examples checked on hardware.

### CODE REVIEW

File: examples/stm32/f1/stm32-smart2/usb_uart/ring.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+extern void ring_init(struct ring *ring, uint8_t *buf, int32_t ring_size);
+
+extern int32_t ring_write_ch(struct ring *ring, uint8_t ch);
+
+extern int32_t ring_write(struct ring *ring, uint8_t *data, int32_t size);
+
+extern int32_t ring_read_ch(struct ring *ring, uint8_t *ch);
+
+extern int32_t ring_read(struct ring *ring, uint8_t *data, int32_t size);
```
Comment (karlp) [inline examples/stm32/f1/stm32-smart2/usb_uart/ring.h]: This isn't how headers work.

**Comment by karlp** [comment]
It's _yet another_ generic F1 board.  What does this PR really contribute?  It's yet more general copy pasta.  It's kinda nice to see the ring buffer impl, but it's _just_ in this one, so we've simply got more divergent forking examples, all varying in varying ways.  Finally, USB from HSI is out of spec, even though it works on _your_ boards.

**Comment by karlp** [comment]
it's not noise noise, it's just hard to maintain it all.  I'm planning on making a new repo that is _just_ blinkys for _anyyyyy_ board people want, so people can test their toolchains and build setups, and then a second repo that has higher level examples, that share code properly, so they can be maintained, and run on more boards there too.

**Comment by baranovskiykonstantin** [comment]
It would be nice.

**Pull request closed without merging**